validphys2/src/validphys/lhaindex.py

```diff
@@ -130,6 +131,12 @@ def parse_info(name):
 
 
 def get_lha_datapath():
+    """Return an existing datapath from LHAPDF, starting from the end.
+    If no path is found to exist, recover the old behaviour and returns the last path.
+    """
+    for lhapath in lhapdf.paths()[::-1]:
+        if Path(lhapath).exists():
```

Comment:
Maybe add the usecase here, because it's not so obvious

Reply:
This is my particular usecase, but tbh what is the point of returning paths that don't exist?

Reply:
It's not so obvious how you could have an lhapdf path that doesn't exist without it being a bigger problem (i.e. meaning the user accidentally removed the LHAPDF folder instead of your usecase).

Having sad that, I agree with your points,but then should we just raise an error instead of recovering the old behaviour?

Reply:
The thing is I don't know why it was done in this way and both lhapdf and this module sometimes do things I don't understand so...

When it is a problem vp will fail anyway 🤷‍♂️

Reply:
Okay so in that case, could you please clarify the usecase in which you would have non-existent lhapdf paths without that being a problem the user would want to be aware of? 

I think just adding this if-statement without that would only make it more black magic

Reply:
I've completed the docstr with some situations in which the check is useful.
I could not find out why do we start counting from the end.

Reply:
Thanks!